### PR TITLE
Add UrlBuilder.parse()

### PR DESCRIPTION
**Pull request by daschult**
I added this method because I need to modify a URL in autorest.java and I need to check if a port already exists. If it doesn't, then I need to add it. java.net.URL doesn't allow me to check if the port already exists, otherwise I would've just used that.

### CODE REVIEW

File: client-runtime/src/main/java/com/microsoft/rest/v2/policy/PortPolicy.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        this.nextPolicy = nextPolicy;
+        this.port = port;
+        this.overwrite = overwrite;
+    }
+
+    @Override
+    public Single<HttpResponse> sendAsync(HttpRequest request) {
+        final UrlBuilder urlBuilder = UrlBuilder.parse(request.url());
+        if (overwrite || urlBuilder.port() == null) {
+            request.withUrl(urlBuilder.withPort(3000).toString());
```
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v2/policy/PortPolicy.java]: default to 3000?

Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/v2/policy/PortPolicy.java]: Copy and paste mistake. Fixed.

**Comment by RikkiGibson** [comment]
It sounds like this would be useful for cases where we have to merge additional parameters into an existing URL. But it also looks like java.net.URL.getPort() returns -1 when the port is not specified in syntax in the URL.

```
    @Test
    public void urlTest() throws Exception {
        java.net.URL url = new java.net.URL("http://microsoft.com");
        assertEquals(80, url.getPort());
    }

    // AssertionError: Expected 80, Actual -1
```

**Comment by daschult** [comment]
@RikkiGibson Yeah, it would be useful for those scenarios. I originally attempted to write my own URL parser, but it just got to be more work than it was worth. I discovered what you mentioned about java.net.URL returning -1 if a port wasn't specified, so I ended up just wrapping it's constructor/parse() method.

**Pull request merged**